ambry-store/src/main/java/com.github.ambry.store/DiskSpaceAllocator.java

```diff
@@ -171,7 +180,15 @@ void allocate(File destinationFile, long sizeInBytes) throws IOException {
       }
       File reserveFile = null;
       if (poolState != PoolState.NOT_INVENTORIED) {
-        reserveFile = reserveFiles.remove(sizeInBytes);
+        if (isSwapSegment) {
+          // attempt to get segment from swap reserve directory
+          reserveFile = swapReserveFiles.remove(sizeInBytes);
+        } else {
+          // attempt to get segment from corresponding store reserve directory
+          if (storeReserveFiles.containsKey(storeId)) {
+            reserveFile = storeReserveFiles.get(storeId).remove(sizeInBytes);
+          }
+        }
       }
       if (reserveFile == null) {
         if (enablePooling) {
```

Comment:
or it might be good to edit all these log message to include info about storeID and isSwapSegment

Reply:
good point, will do.